src/index.js

```diff
@@ -113,9 +113,13 @@ function validEncode (name, buf) {
  * @throws {Error} Will throw if the encoding is not supported
  */
 function encoding (nameOrCode) {
+  // @ts-ignore
```

Comment:
It might be worth to add the reason for ignoring next to the `@ts-ignore`

Reply:
I think it would be better & clearer to cast types (instead of adding these ignores) as [following example illustrates](https://www.typescriptlang.org/play?target=5&useJavaScript=true#code/PQKhAIChwgBAXAngBwKYBNUDNwG8DkADPgD74Dsp+AnPgL7gBCAhgM6oDCA9ptHEmkw4CAIzaoATFTHsAHNPEBGYgxbsAcswC2qPuAQoM2PGs49UJU5p2rx11AHkATt14xgkSKBB7YyZk7aJnbaji7mDAB2oc6uujD6TqjwAK5OkazB7HR8HlgpkQDG8ACWXJHgqEU8JZEA5uAAFNE6seYAlHjQ4IXlrPDgJejgALzg3voCqFmo9uAAZEzicQwgwM0x4Zjt3SU4jb0Z8MyR8KwAdC2orADaQwC6nbjd4OBJqek9fcenF1e3D26DFQABt2IN9od+iczudepgAehHl1Xq93mkKlCfrD4dc7kigZUwdNnqjwPAABZOLgAd3AkVQdIAok5qU5GgADACqGRSyGQXCc8AwlWq6FqdQAXOAACS4K5tTB0Dk7V45HJeMCTQx4eWhViSgBKqF6TnQAB4rKEADRLdgAPltuIN4GNpotpjittM9roqw8wGAk1YAFoSnVIoLdFivkcYaxPJBCiC2JlTF0NUA)

```js
/**
 * @param {BaseNameOrCode} nameOrCode
 * @returns {Base}
 */
function encoding (nameOrCode) {
  const id = /** @type {BaseName & BaseCode} */(nameOrCode)
  if (constants.names[id]) {
    return constants.names[id]
  } else if (constants.codes[id]) {
    return constants.codes[id]
  } else {
    throw new Error(`Unsupported encoding: ${nameOrCode}`)
  }
}
```